pkg/controllers/deployment/deployment_controller.go

```diff
@@ -227,6 +211,11 @@ func (c *deploymentController) sync(ctx context.Context, syncContext factory.Syn
 		return err
 	}
 
+	if len(foundConditions) > 0 {
+		klog.V(4).Infof("Retrying because conditions: %s", spew.Sdump(foundConditions))
+		return factory.SyntheticRequeueError
+	}
```

Comment:
I missed this bit with the other controllers also. Shouldn't this be assumed by the events that you connect the controller to, instead of looping on a errors?

Reply:
@stlaz i think retrying faster when there are degraded conditions makes sense, but now that most of the controllers are driven by listers I think it is fine if we remove this?

Reply:
the events you attach to have always been driven by informers, correct? I would assume that in most cases you would just see that an object your controller watches changed and a new loop is started to handle the event.

I can think of just one exception which is the KAS grabbing the metadata configmap and performing rollouts that are slow and we're not watching the static pods, but in that case, time resync should be used (and we used it everywhere now).

Reply:
right, initially, some of these were driven by clients which means you wanted to retry when something bad happened... as most of these are now listers driven, I think the risk is low and the periodic resync should handle wedged controller case (if there are any).

Reply:
(removed)